File: sword-means-offer-2/029/main.py
# -*- coding: UTF-8 -*-
# @Time     : 2022/01/20 23:07
# @Author   : Ranshi
# @File     : 123.py
# @Doc      : 剑指 Offer II 029. 排序的循环链表
from pyal.container import ListNode as Node


class Solution:
    def insert(self, head: "Node", insertVal: int) -> "Node":
        if not head:
            node = Node(val=insertVal)
            node.next = node
            return node
        pre, node = head, Node(val=insertVal)
        while True:
            if (
                pre.val <= insertVal <= pre.next.val
                or (pre.next.val < pre.val and (insertVal >= pre.val or insertVal <= pre.next.val))
                or pre.next == head
            ):
                pre.next, node.next = node, pre.next
                break
            pre = pre.next
        return head

File: sword-means-offer-2/029/main.go
package main

// @Time     : 2022/09/15 20:33
// @Author   : Ranshi
// @File     : main.go
// @Doc      : 剑指 Offer II 029. 排序的循环链表

type Node struct {
	Val  int
	Next *Node
}

func insert(head *Node, x int) *Node {
	if head == nil {
		node := &Node{Val: x}
		node.Next = node
		return node
	}
	pre, node := head, &Node{Val: x}
	for {
		if pre.Val <= x && x <= pre.Next.Val || (pre.Next.Val < pre.Val && (x >= pre.Val || x <= pre.Next.Val)) || pre.Next == head {
			pre.Next, node.Next = node, pre.Next
			break
		}
		pre = pre.Next
	}
	return head
}
